great_expectations/cli/datasource.py

```diff
@@ -1049,11 +1048,28 @@ def _get_batch_kwargs_from_generator_or_from_file_path(context, datasource_name,
 
                 if reader_method is not None:
                     batch_kwargs["reader_method"] = reader_method
+                    if isinstance(datasource, SparkDFDatasource) and reader_method == "csv":
+                        header_row = click.confirm(
+                            "\nDoes this file contain a header row?",
+                            default=True
+                        )
+                        batch_kwargs["reader_options"] = {
+                            "header": header_row
+                        }
                     batch = datasource.get_batch(batch_kwargs=batch_kwargs)
                     break
         else:
             # TODO: read the file and confirm with user that we read it correctly (headers, columns, etc.)
             try:
+                batch_kwargs["reader_method"] = reader_method
+                if isinstance(datasource, SparkDFDatasource) and reader_method == "csv":
+                    header_row = click.confirm(
+                        "\nDoes this file contain a header row?",
+                        default=True
+                    )
+                    batch_kwargs["reader_options"] = {
+                        "header": header_row
+                    }
```

Comment:
Wonderful! I wonder how much effort would be involved in writing a test for this.